### PR TITLE
Added Interface to all contracts

### LINKED ISSUE #394: All contracts should inherit their own interface

### PR DESCRIPTION
**Pull request by adi44**
fixes #394

### CODE REVIEW

File: contracts/Core/Parameters.sol [hunk 1]
```diff
@@ -112,12 +113,80 @@ contract Parameters is ACL, Constants {
...
         return (uint32(block.number) / (epochLength));
     }
 
-    function getState() external view returns (uint8) {
+    function getState() external view override returns (uint8) {
         uint8 state = uint8(((block.number) / (epochLength / NUM_STATES)) % (NUM_STATES));
         return (state);
     }
+
+    function getEpochLength() external view override returns (uint16) {
```
Comment (abhishekvispute) [inline contracts/Core/Parameters.sol:125]: hey is there a reason to add these getters ?

Comment (adi44) [reply contracts/Core/Parameters.sol]: I have removed them.

**Review by abhishekvispute — APPROVED** [review]

File: contracts/Core/interface/IVoteManager.sol [hunk 2]
```diff
@@ -8,20 +8,20 @@ interface IVoteManager {
 
     function reveal(
```
Comment (hrishikeshio) [inline contracts/Core/interface/IVoteManager.sol:9]: It seems there are lot of definitions in interface that are not being used anywhere?
These should be removed to save space.

Comment (adi44) [reply contracts/Core/interface/IVoteManager.sol:9]: @hrishikeshio yeah, Few getters are not being used.

Comment (hrishikeshio) [reply contracts/Core/interface/IVoteManager.sol:9]: not just getters.
Where are we using reveal() ? many write functions are also not being used. please remove.

File: contracts/Core/interface/IVoteManager.sol [hunk 3]
```diff
@@ -4,24 +4,24 @@ pragma solidity ^0.8.0;
 import "../../lib/Structs.sol";
 
 interface IVoteManager {
-    function commit(uint32 epoch, bytes32 commitment) external;
+    //function commit(uint32 epoch, bytes32 commitment) external;
 
-    function reveal(
+    /* function reveal(
```
Comment (hrishikeshio) [inline contracts/Core/interface/IVoteManager.sol:9]: why comment? remove.

Comment (adi44) [reply contracts/Core/interface/IVoteManager.sol]: Done!

File: contracts/Core/interface/IAssetManager.sol [hunk 4]
```diff
@@ -2,7 +2,7 @@
 pragma solidity ^0.8.0;
 
 interface IAssetManager {
-    function getAssetType(uint8 id) external view returns (uint8);
+    //function getAssetType(uint8 id) external view returns (uint8);
```
Comment (hrishikeshio) [inline contracts/Core/interface/IAssetManager.sol:5]: remove

File: contracts/Core/interface/IAssetManager.sol [hunk 5]
```diff
@@ -16,7 +16,7 @@ interface IAssetManager {
             string memory url
         );
 
-    function getCollection(uint8 id)
+    /*function getCollection(uint8 id)
```
Comment (hrishikeshio) [inline contracts/Core/interface/IAssetManager.sol:19]: remove

File: contracts/Core/interface/IAssetManager.sol [hunk 6]
```diff
@@ -25,11 +25,11 @@ interface IAssetManager {
             uint8[] memory jobIDs,
             uint32 aggregationMethod,
             string memory name
-        );
+        );*/
 
     function getNumActiveAssets() external view returns (uint8);
 
-    function getActiveAssets() external view returns (uint8[] memory);
+    //function getActiveAssets() external view returns (uint8[] memory);
```
Comment (hrishikeshio) [inline contracts/Core/interface/IAssetManager.sol:32]: remove

File: contracts/Core/interface/IBlockManager.sol [hunk 7]
```diff
@@ -14,41 +14,34 @@ interface IBlockManager {
     // note that only one staker or no stakers selected in each iteration.
     // stakers elected in higher iterations can also propose hoping that
     // stakers with lower iteration do not propose for some reason
-    function propose(
+    /*function propose(
```
Comment (hrishikeshio) [inline contracts/Core/interface/IBlockManager.sol:17]: remove all commented code

File: contracts/Core/interface/IAssetManager.sol (hunk 4 above)
Comment (adi44) [reply contracts/Core/interface/IAssetManager.sol:5]: this is removed

**Review by hrishikeshio — APPROVED** [review]

**Review by abhishekvispute — APPROVED** [review]

**Pull request merged**